### PR TITLE
[not for merge] Add parameter for leak tracking to test widgets.

### PR DESCRIPTION
**Pull request by polina-c**
Contributes to https://github.com/flutter/flutter/issues/135856

### CODE REVIEW

**Comment by goderbauer** [comment]
Looks like this is breaking a lot of checks (including google3). Can you take a look?

File: packages/flutter_test/lib/src/widget_tester.dart [hunk 1]
```diff
@@ -116,6 +117,11 @@ E? _lastWhereOrNull<E>(Iterable<E> list, bool Function(E) test) {
 /// If the [tags] are passed, they declare user-defined tags that are implemented by
 /// the `test` package.
 ///
+/// The argument [experimentalLeakTesting] is experimental and is not recommended
+/// for use outside of Flutter Framework.
+/// [experimentalLeakTesting] is noop.
```
Comment (goderbauer) [inline packages/flutter_test/lib/src/widget_tester.dart:122]: nit: sentences should start grammatically correct with a capital letter: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#use-correct-grammar

File: packages/flutter_test/lib/src/widget_tester.dart [hunk 2]
```diff
@@ -135,7 +141,10 @@ void testWidgets(
   TestVariant<Object?> variant = const DefaultTestVariant(),
   dynamic tags,
   int? retry,
+  LeakTesting? experimentalLeakTesting,
 }) {
+  // TODO(polina-c): enable leak tracking and update doc comment for `experimentalLeakTesting`.
+  // https://github.com/flutter/flutter/issues/135856
```
Comment (goderbauer) [inline packages/flutter_test/lib/src/widget_tester.dart:147]: Why not just implement this in this PR? Its odd to just add a parameter that doesn't do anything.

File: packages/flutter_test/lib/src/widget_tester.dart [hunk 3]
```diff
@@ -116,6 +117,11 @@ E? _lastWhereOrNull<E>(Iterable<E> list, bool Function(E) test) {
 /// If the [tags] are passed, they declare user-defined tags that are implemented by
 /// the `test` package.
 ///
+/// The argument [experimentalLeakTesting] is experimental and is not recommended
+/// for use outside of Flutter Framework.
+/// [experimentalLeakTesting] is noop.
+/// Use the method `testWidgetsWithLeakTracking` for leak free tests.
```
Comment (goderbauer) [inline packages/flutter_test/lib/src/widget_tester.dart:123]: This is a pkg:flutter internal method that we don't expose to the public, no? We probably shouldn't reference this in our public docs as "nobody" on the outside can really use this...

File: packages/flutter_test/lib/src/widget_tester.dart [hunk 4]
```diff
@@ -9,6 +9,7 @@ import 'package:flutter/material.dart' show Tooltip;
 import 'package:flutter/rendering.dart';
 import 'package:flutter/scheduler.dart';
 import 'package:flutter/services.dart';
+import 'package:leak_tracker_flutter_testing/leak_tracker_flutter_testing.dart';
```
Comment (goderbauer) [inline packages/flutter_test/lib/src/widget_tester.dart:12]: Would you have to update the pubspec.yaml to declare this as a public dependency?

File: packages/flutter_test/lib/src/widget_tester.dart (hunk 2 above)
Comment (polina-c) [reply packages/flutter_test/lib/src/widget_tester.dart:147]: I need to update leaking tests to set leak tracking off first, using this parameter.

File: packages/flutter_test/lib/src/widget_tester.dart (hunk 3 above)
Comment (polina-c) [reply packages/flutter_test/lib/src/widget_tester.dart:123]: 1. The doc is saying: `is not recommended for use outside of Flutter Framework.`
2. This dos is for FF developers, who are supposed to use this parameter. 
3. It is temporary state: testWidgetsWithLeakTracking will be cleaned up very soon and the doc will be updated.

File: packages/flutter_test/lib/src/widget_tester.dart (hunk 4 above)
Comment (polina-c) [reply packages/flutter_test/lib/src/widget_tester.dart:12]: It is already here: https://github.com/flutter/flutter/pull/137069

**Comment by polina-c** [comment]
> Looks like this is breaking a lot of checks (including google3). Can you take a look?

Yes, I need to figure it out.

**Comment by polina-c** [comment]
Keeping this PR open to be example for web compiler bug.

File: packages/flutter_test/lib/src/widget_tester.dart (hunk 1 above)
Comment (polina-c) [reply packages/flutter_test/lib/src/widget_tester.dart]: Resolved in https://github.com/flutter/flutter/pull/137655

**Comment by goderbauer** [comment]
Can this be closed in favor of https://github.com/flutter/flutter/pull/138057 (or https://github.com/flutter/flutter/pull/137655)?

**Comment by polina-c** [comment]
Closing in favor of https://github.com/flutter/flutter/pull/137655

**Pull request closed without merging**